src/grids.jl

```diff
@@ -14,6 +14,14 @@ struct RegularCartesianGrid{T<:AbstractFloat, R<:AbstractRange} <: Grid{T}
     Nx::Int
     Ny::Int
     Nz::Int
+    # Halo size in (x,y,z).
+    Hx::Int
+    Hy::Int
+    Hz::Int
+    # Total number of grid points (including halo regions).
+    Tx::Int
+    Ty::Int
+    Tz::Int

```

Comment:
should we `NamedTuple` this stuff? ie `N=(x=16, y=16, z=16)` so that `N.x = 16`? Maybe not now but something to ponder on.

Reply:
Yeah that might be a good idea. There's also the issue that `Tx, Ty, Tz` is also used for the number of CUDA threads per block, but I figured they're in different structs so it's OK for now.

Reply:
I think that is a legitimate concern — harder to read the code and increases potential for bugs. Though less clean, we could use NH and NT for number of halo points and number of total points... or *look to unicode*

Reply:
> @johncmarshall54: T and H are very often used for temperatures and depth. Source of confusion? Probably not. But other symbols might be better to represent threads etc etc.

Reply:
We use `Tx, Ty, Tz` for threads but that's going to change soon. Seems like we'll end up with something close to `layout.threads.x`, etc. from PR #249. So there probably won't be a conflict.

`T` and `H` are used for temperature but I guess right now the halo size and total number of grid points are very internal and only used in one or two places so I'll keep them as is because I'm not sure of a better name for them.